### ISSUE TITLE
No Pin to Panel option for Ambient Noise

### ISSUE THREAD

**Issue by aldoiljazi** [solus-project/budgie-desktop#750]
There is no Pin to Panel Ambient Noise option.

Screenshot:

![ambient noise bug](https://cloud.githubusercontent.com/assets/12599754/20862376/478bdbbc-b9b1-11e6-8196-1da75d799791.png)

**Comment by Vistaus** [comment]
Yeah, I've also noticed there are quite a few apps where the option's missing.

**Comment by ckrzen** [comment]
I find that if I add the *.desktop file to:

~/.local/share/applications/

AND(in some cases):

right-click on the *.desktop entry | select the the permissions tab | and
check the "Allow executing file as program" tick-box

that that then exposes the "Pin to panel" option on the Budgie menu.


HTH!

**Comment by feskyde** [comment]
> Yeah, I've also noticed there are quite a few apps where the option's missing.

Telling us which apps could be more useful 😉

**Comment by Vistaus** [comment]
You're right, I'm sorry.

From the top of my head (which I encountered):
-Dolphin (the KDE File Manager)
-Vivaldi (web browser)
-Vivaldi apps (using Vivaldi's --app argument in a launcher)
-Xfe (file manager)
-Xfw (text editor part of the Xfe suite)

I'll add more to the list when I remember or encounter more.

**Comment by Vistaus** [comment]
Another one to add to the previous list: Gramps.

**Comment by ikeydoherty** [comment]
If they don't provide a WM_CLASS there's literally nothing I can do. Ambient is one of them situations. Budgie can only act on the information provided by the app. If they omit the information, or they break it - I have nothing to associate to a .desktop file

**Comment by ikeydoherty** [comment]
TL;DR It's not a magician. It makes informed decisions.

**Comment by ikeydoherty** [comment]
This change will help a bit though:

https://github.com/budgie-desktop/budgie-desktop/commit/da75a4e484b67a1f9dbe1688a81c7e7661ac4a98

**Comment by JoshStrobl** [comment]
Someone please confirm as to whether or not the latest Budgie resolves this issue.

**Comment by aldoiljazi** [comment]
Now when I launch Ambient Noise, I don't see any window launching. I do listen to the "noise" though.

Probably a separate bug?

**Comment by baimafeima** [comment]
Yeah, I don't see any window launching either. The only way to shut it down is via the terminal.

**Comment by Vistaus** [comment]
Another one to add to the list: QupZilla.

**Comment by Radagast** [comment]
With the Panel or Panel Dock (Icon Task List I guess), I cannont pin Firefox Nightly.

**Comment by ikeydoherty** [comment]
OK as I've said before the .desktop file needs to actually match up to the wmclass - make sure firefox nightly actually does this

**Comment by joebonrichie** [comment]
This is not a budgie bug, I fixed `anoise` itself by adding 'StartupWMClass' to it's desktop file.
Resolved here: https://dev.solus-project.com/R357:b23a3e6385fc9a8e8083d9363c5ebc41201d9ef8

This can be closed.

**Comment by joebonrichie** [comment]
Qupzilla fixed
https://dev.solus-project.com/R2775:55cd7eb902b012cc7dd508c446a7a8e76680f89c

**Comment by joebonrichie** [comment]
Vivaldi, dolphin, gramps, all seem to be fine. We don't seem to have xfe in the repos.
If anymore applications have this bug please report on phabricator (https://dev.solus-project.com/) instead and I can fix it and also send a patch upstream.

**Comment by JoshStrobl** [comment]
@joebonrichie Except **not all** Budgie users use Solus so it'd be better if you file an issue upstream directly with the developers of each app rather than catering specifically to Solus.

**Comment by joebonrichie** [comment]
Yeah don't worry, i've sent pull requests to QupZilla and ReText so far which have been merged. Still need to do Kdenlive and Anoise.

**Comment by nessunluogo** [comment]
I had the same problem with PhpStorm. Fixed editing the StartupWMClass to desktop file (in ~/.local/share/applications/).
To get the right value, just run in your terminal:
`xprop WM_CLASS`
then use the + pointer to click the window. You'll get two strings, use the second ad StartupWMClass.

**Comment by JoshStrobl** [comment]
Confirmed this is resolved with latest IconTasklist + Abomination implementation in git.
